src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt

```diff
@@ -78,27 +87,45 @@ fun zipDirectory(directory: File): ByteArray {
     return byteOutput.toByteArray()
 }
 
-class ResultReceiver() : CompilerFleetMessageReceiver() {
-    override fun processMessage(message: PubsubMessage) {
+fun getTaskId(data: ByteArray): String {
+    val messageDigest = MessageDigest.getInstance("SHA-1")
+
+    return String(messageDigest.digest(data))
+}
+
+private val LOGGER = LoggerFactory.getLogger("ResultReceiver")
+
+class ResultReceiver(
+        private val outputFile: File,
+        private val expectedTaskId: String
+) : CompilerFleetMessageReceiver() {
+    override fun processMessage(message: PubsubMessage): Boolean {
         val result = CompilerFleet.CompilerFleetResult.parseFrom(message.data)
 
-        println(result.taskId)
-        println(String(result.resultBytes.toByteArray()))
+        if (result.taskId != expectedTaskId) {
+            LOGGER.info("Task ids don't match: \nexpected:{}, \nactual:{}", expectedTaskId, result.taskId)
+            return false
+        }
+
+        FileUtils.writeByteArrayToFile(outputFile, result.resultBytes.toByteArray())
+        return true
```

Comment:
Please add info logging indicating that the result was successfully received and written, otherwise it just hangs silently. Or maybe call System.exit(0) ? After receiving the result we can't do anything meaningful anyway.

Reply:
Is it okay exit(0) and then return? Looks weird :)

Reply:
C'est la vie

Reply:
Done